rcl/include/rcl/parameter.h

```diff
@@ -0,0 +1,177 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCL__PARAMETER_H_
+#define RCL__PARAMETER_H_
+
+#if __cplusplus
+extern "C"
+{
+#endif
+
+/// rcl_parameter_client_set_<TYPE> adds the parameter key, value pair to the
+/// pending set_parameters_request
+// CTYPE is a macro to sanitize human-readble type names to valid c types
+#define RCL_DECLARE_SET_PARAMETER(TYPE) \
+RCL_PUBLIC \
+RCL_WARN_UNUSED \
+rcl_ret_t \
+rcl_parameter_set_ ## TYPE ## ( \
+  rcl_interfaces__msg__Parameter * parameter, const char * name, CTYPE(TYPE) value); \
+
+RCL_PUBLIC
+RCL_WARN_UNUSED
+rcl_ret_t
+rcl_parameter_set_bool(
+  rcl_interfaces__msg__Parameter * parameter, const char * parameter_name, bool value);
+
+// Or, with the above macro, it would simply be RCL_DECLARE_SET_PARAMETER_REQUEST(bool)
```

Comment:
If there are only five variants (that's how many I count anyways) then maybe the macro isn't necessary for the function prototypes. In the header I think it would be best to avoid MACRO's for documentation purposes. In the C files where common lines of code should automatically distribute fixes (fix one line, fix all the set or get functions), I think the macro's make more sense.